gemrb/plugins/SDLAudio/SDLAudio.cpp

```diff
@@ -252,24 +319,30 @@ Holder<SoundHandle> SDLAudio::Play(const char* ResRef, unsigned int channel,
 		*length = time_length;
 	}
 
-	Mix_VolumeChunk(chunk, MIX_MAX_VOLUME * GetVolume(channel) / 100);
-
-	// play
-	int chan = -1;
-	if (flags & GEM_SND_SPEECH) {
-		chan = 0;
-	}
+	Mix_VolumeChunk(chunk, MIX_MAX_VOLUME * (GetVolume(channel) * volume / 10000.0f));
 
 	std::lock_guard<std::recursive_mutex> l(OurMutex);
-	
-	chan = Mix_PlayChannel(chan, chunk, 0);
+
+	chan = Mix_PlayChannel(chan, chunk, loop);
 	if (chan < 0) {
 		print("error playing channel");
 		return Holder<SoundHandle>();
 	}
 
-	// TODO
-	return Holder<SoundHandle>();
+	// Positional audio
+	if (!(flags & GEM_SND_RELATIVE))
+	{
+		int x = listenerPos.x - XPos;
+		int y = listenerPos.y - YPos;
+		double angle = atan2(y, x);
+		if (angle < 0) {
+			angle += 360;
+		}
+		uint8_t distance = std::min(static_cast<int32_t>(sqrt(x * x + y * y) / AUDIO_DISTANCE_ROLLOFF_MOD), 255);
+		Mix_SetPosition(chan, angle * 180 / M_PI - 90, distance);
+	}
```

Comment:
This chunk is the same as in SDLAudioSoundHandle::SetPos, so I suggest you extract it to a common private method.

Reply:
These are two different classes tho. Should I just make some separate function in the file or some public static method in SDLAudio?

Reply:
File static sounds fine, since you only have a few params.

Reply:
Done

Reply:
Thanks! Is this PR done or are you still working on something (like GEM_SND_VOL_AMBIENTS)?

Reply:
I think it's done. No plans for GEM_SND_VOL_AMBIENTS for now